### PR TITLE
Code maintenance and cleanup (mostly Java 7 stuff)

### PR DESCRIPTION
**Pull request by VibyJocke**
Replaced legacy constructs with modern practices (try-with-resources, foreach, diamond operator)
Removed redundant modifiers (public/static/final interface constants, etc.), casts (Integer -> int) and calls (toString() on Strings).
Changes limited to the server maven module.

### CODE REVIEW

**Comment by ketan** [comment]
@varshavaradarajan Can you review and merge this?

File: server/src/com/thoughtworks/go/server/gadget/GadgetDataSource.java [hunk 1]
```diff
@@ -58,7 +58,7 @@
 
     GadgetOauthAccessTokenDTO saveOauthAccessToken(Map attributes);
 
-    public static class GadgetOauthClientDTO {
+    public class GadgetOauthClientDTO {
```
Comment (varshavaradarajan) [inline server/src/com/thoughtworks/go/server/gadget/GadgetDataSource.java]: @VibyJocke,  you can remove the access modifier 'public' too.

File: server/src/com/thoughtworks/go/server/service/BackupService.java [hunk 2]
```diff
@@ -149,33 +149,21 @@ private void backupVersion(File backupDir) throws IOException {
...
-            }
         }
     }
 
     private void backupConfig(File backupDir) throws IOException {
         String configDirectory = systemEnvironment.getConfigDir();
-        ZipOutputStream configZip = null;
-        try {
-            configZip = new ZipOutputStream(new BufferedOutputStream(new FileOutputStream(new File(backupDir, CONFIG_BACKUP_ZIP))));
+        try (ZipOutputStream configZip = new ZipOutputStream(new BufferedOutputStream(new FileOutputStream(new File(backupDir, CONFIG_BACKUP_ZIP))));) {
```
Comment (varshavaradarajan) [inline server/src/com/thoughtworks/go/server/service/BackupService.java]: the semi-colon is unnecessary. :)

File: server/src/com/thoughtworks/go/server/service/BackupService.java [hunk 3]
```diff
@@ -251,14 +239,8 @@ protected void handleFile(File file, int depth, Collection results) throws IOExc
             return;
         }
         zipStream.putNextEntry(new ZipEntry(fromRoot(file)));
-        BufferedInputStream in = null;
-        try {
-            in = new BufferedInputStream(new FileInputStream(file));
+        try (BufferedInputStream in = new BufferedInputStream(new FileInputStream(file));) {
```
Comment (varshavaradarajan) [inline server/src/com/thoughtworks/go/server/service/BackupService.java]: The semi-colon is unnecessary.

**Comment by varshavaradarajan** [comment]
@VibyJocke, two things -
Please sign the Contributor License Agreement: https://www.go.cd/contributor-license-agreement/
Please update the copyright in these files. To get an idea, you can look at https://github.com/gocd/gocd/blob/653cd70d096093329b3969d5dff1d4d450c93da0/common/src/com/thoughtworks/go/domain/materials/git/GitCommand.java#L2
Everything else looks good. 👍

Keep an eye out for failing tests on build.go.cd's PR pipelines.

**Comment by VibyJocke** [comment]
Thank you for the feedback @varshavaradarajan 
About the copyright, I'll make sure to update them, but should I also add them to files where they are missing?

**Comment by varshavaradarajan** [comment]
Yes please.

**Comment by VibyJocke** [comment]
@varshavaradarajan I think I'm done now :) where can I see the PR pipeline test results?

**Comment by varshavaradarajan** [comment]
https://build.go.cd/go/pipelines Pipeline: create-maven-release-PR

**Comment by varshavaradarajan** [comment]
@VibyJocke, can you squash the 5 commits into one?

**Comment by VibyJocke** [comment]
Squashed and synced with upstream/master.

**Comment by varshavaradarajan** [comment]
@VibyJocke, can you please resolve the merge conflicts?

**Comment by VibyJocke** [comment]
Synced with upstream/master

File: server/src/com/thoughtworks/go/config/ConfigConverter.java [hunk 4]
```diff
@@ -422,7 +422,7 @@ public JobConfig toJobConfig(CRJob crJob) {
         else {
             Integer count = crJob.getRunInstanceCount();
             if (count != null)
-                jobConfig.setRunInstanceCount(count);
+                jobConfig.setRunInstanceCount((int) count);
```
Comment (varshavaradarajan) [inline server/src/com/thoughtworks/go/config/ConfigConverter.java]: Didn't you remove these as redundant?

**Comment by VibyJocke** [comment]
Missed to commit the resolved ConfigConverter from the merge. It's in now.

**Pull request merged**